domain/src/test/java/jinsu/antilog/domain/CalculatorTest.kt

```diff
@@ -0,0 +1,56 @@
+package jinsu.antilog.domain
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Assert.*
+import org.junit.Test
+
+class CalculatorTest {
+    @Test
+    fun givenExpression_whenCalculate_then10() {
+        // given : "2 + 3 * 4 / 2" 수식이 주어지고
+        val expressionString = "2 + 3 * 4 / 2"
+        // when : 계산기로 계산을 했을 때
+        val calculatedValue = Calculator().calculate(expressionString)
+        // then : 10.0 이라는 결과가 나온다.
+        assertThat(calculatedValue).isEqualTo(10.0)
+    }
+
+    @Test
+    fun givenNullExpressionString_whenCalculate_thenThrowIllegalArgumentException() {
+        // given : null 값의 수식이 주어지고
+        val expressionString: String? = null
+        // when : 계산기로 계산을 했을 때
+        // then : IllegalArgumentException 이 발생한다.
+        assertThrows(IllegalArgumentException::class.java){
+            Calculator().calculate(expressionString)
+        }.also {
+            assertThat(it.message).isEqualTo("전달 받은 문자열이 Null 이거나 공백일 수 없습니다.")
+        }
```

Comment:
also를 통해 한 번에 합쳐서 진행한 이 형태는 when, then절이 합쳐진 모양새입니다.
exception 인스턴스를 받아서 따로 처리하면, 명시적으로 "~~를 했을 때 ~~라는 에러가 일어난다" 라고 작성할 수 있습니다.
```suggestion
        // when : 계산기로 계산을 했을 때
        val actualException = kotlin.runCatching { Calculator().calculate(expressionString) }.exceptionOrNull()
        
        // then : IllegalArgumentException 이 발생한다.
        assertThat(actualException).isInstanceOf(IllegalArgumentException::class.java)
        assertThat(actualException).hasMessageThat().contains("전달 받은 문자열이 Null 이거나 공백일 수 없습니다.")
```